### PR TITLE
Problem: TX not sealed / stored on the enclave side (CRO-229)

### PR DESCRIPTION
**Pull request by tomtau**
Solution:
- added lookups of the stored sealed transactions in enclave app wrapper + unsealing routines in the enclave code
- extended the `SGX_TEST=1` unit test execution

### CODE REVIEW

File: enclave/src/lib.rs [hunk 1]
```diff
@@ -47,6 +47,51 @@ fn check_chain_info(chain_info: *const u8, chain_info_len: usize) -> Option<Chai
     }
 }
 
+#[inline]
```
Comment (devashishdxt) [inline enclave/src/lib.rs]: Do we want to make this function inline?

File: enclave/src/lib.rs [hunk 2]
```diff
@@ -47,6 +47,51 @@ fn check_chain_info(chain_info: *const u8, chain_info_len: usize) -> Option<Chai
...
+    };
+    let otx = TxWithOutputs::decode(&mut unsealed_data.get_decrypt_txt());
+    // TODO: check decoded txid against unsealed_data.get_additional_txt?
+    match otx {
+        Ok(tx) => Some(tx),
+        _ => None,
+    }
+}
+
+#[inline]
```
Comment (devashishdxt) [inline enclave/src/lib.rs]: Do we want to make this function inline?

File: enclave/src/lib.rs (hunk 1 above)
Comment (tomtau) [reply enclave/src/lib.rs]: Yes, I think both are small utilities that would make sense to be inline -- they were extracted out to functions for reuse between two ecall functions (I'm still thinking whether to have multiple ecall functions or one) that unseal inputs and a bit better readability

File: enclave/src/lib.rs (hunk 2 above)
Comment (tomtau) [reply enclave/src/lib.rs]: https://github.com/crypto-com/chain-tx-enclave/pull/11#discussion_r310433345

File: enclave/src/lib.rs (hunk 1 above)
Comment (leejw51) [reply enclave/src/lib.rs]: inline function makes code bigger, do you need inline for enclave compatibility?

File: app/src/enclave_u/mod.rs [hunk 3]
```diff
@@ -76,15 +75,15 @@ pub fn check_initchain(eid: sgx_enclave_id_t, chain_hex_id: u8) -> Result<(), ()
 pub fn check_transfertx(
     eid: sgx_enclave_id_t,
     txaux: TxAux,
-    txins: Vec<TxWithOutputs>,
+    txins: Vec<Vec<u8>>,
```
Comment (leejw51) [inline app/src/enclave_u/mod.rs]: about about 
type Vec<u8>= Tx
Vec<Tx>

Comment (tomtau) [reply app/src/enclave_u/mod.rs]: Not possible -- `app/src/enclave_u` = untrusted code which cannot unseal the data (as it's not in the enclave)

**Pull request merged**